build-rpi-raspbian/first-boot.sh

```diff
@@ -44,9 +44,25 @@ elif grep -q "openHABian" /etc/wpa_supplicant/wpa_supplicant.conf; then
   echo "$(timestamp) [openHABian] Setting up Wi-Fi connection... OK"
 else
   echo -n "$(timestamp) [openHABian] Setting up Wi-Fi connection... "
+
+  # check the user input for the country code
+  # check: from the start of line, the uppercased input must be followed by a whitespace
+  if grep -q "^${wifi_country^^}\s" /usr/share/zoneinfo/zone.tab; then
+    wifi_country=${wifi_country^^}
+  else
+    echo "${wifi_country} is not a valid country code found in /usr/share/zoneinfo/zone.tab"
+    echo "Defaulting to US"
+    wifi_country=US
+  fi
+
   echo -e "# config generated by openHABian first boot setup" > /etc/wpa_supplicant/wpa_supplicant.conf
-  echo -e "country=US\\nctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev\\nupdate_config=1" >> /etc/wpa_supplicant/wpa_supplicant.conf
+  echo -e "country=$wifi_country\\nctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev\\nupdate_config=1" >> /etc/wpa_supplicant/wpa_supplicant.conf
   echo -e "network={\\n\\tssid=\"$wifi_ssid\"\\n\\tpsk=\"$wifi_psk\"\\n\\tkey_mgmt=WPA-PSK\\n}" >> /etc/wpa_supplicant/wpa_supplicant.conf
+  
+  echo -e "# config generated by openHABian first boot setup"
+  sed -i '$ d' /etc/default/crda # removes the last line
+  echo -e "REGDOMAIN=$wifi_country" >> /etc/default/crda # adds the country code
```

Comment:
I don't like this method. It is doomed to create issues, I guarantee it...
Go with this instead: https://github.com/openhab/openhabian/blob/master/functions/system.sh#L245